src/pkg/repository/repository.go

```diff
@@ -117,6 +121,50 @@ func (r repository) GetID() string {
 	return r.ID
 }
 
+// New constructs a repository that can be used to Initialize or Connect a repo instance.
+func New(
+	ctx context.Context,
+	acct account.Account,
+	s storage.Storage,
+	opts control.Options,
+	configFileRepoID string,
+) (repo *repository, err error) {
+	ctx = clues.Add(
+		ctx,
+		"acct_provider", acct.Provider.String(),
+		"acct_id", clues.Hide(acct.ID()),
+		"storage_provider", s.Provider.String())
+
+	bus, err := events.NewBus(ctx, s, acct.ID(), opts)
+	if err != nil {
+		return nil, clues.Wrap(err, "constructing event bus").WithClues(ctx)
+	}
+
+	repoID := configFileRepoID
+	if len(configFileRepoID) == 0 {
+		repoID = newRepoID(s)
+	}
+
+	bus.SetRepoID(repoID)
+
+	r := repository{
+		ID:      repoID,
+		Version: "v1",
+		Account: acct,
+		Storage: s,
+		Bus:     bus,
+		Opts:    opts,
+	}
+
+	if !r.Opts.DisableMetrics {
+		bus.SetRepoID(r.ID)
+	}
+
+	r.Bus.Event(ctx, events.RepoInit, nil)
```

Comment:
This is gonna log a `RepoInit` event for connect, instead of `RepoConnect`.

Reply:
whoops, thank you!